src/inc/til/u8u16convert.h

```diff
@@ -299,56 +470,120 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
+    // - Takes a UTF-16 string and performs the conversion to ANSI. NOTE: The function relies on getting complete UTF-16 characters at the string boundaries.
     // Arguments:
+    // - codepage - id of the codepage to be used for the conversion
     // - in - UTF-16 string to be converted
-    // - out - reference to the resulting UTF-8 string
+    // - out - reference to the resulting ANSI string
     // Return Value:
     // - S_OK          - the conversion succeded
+    // - E_INVALIDARG  - the codepage id is invalid
     // - E_OUTOFMEMORY - the function failed to allocate memory for the resulting string
     // - E_ABORT       - the resulting string length would exceed the upper boundary of an int and thus, the conversion was aborted before the conversion has been completed
     // - E_UNEXPECTED  - an unexpected error occurred
     template<class inT, class outT>
     [[nodiscard]] typename std::enable_if<std::is_same<typename inT::value_type, wchar_t>::value && std::is_same<typename outT::value_type, char>::value, HRESULT>::type
-    u16u8(const inT in, outT& out) noexcept
+    u16a(const unsigned int codepage, const inT in, outT& out) noexcept
     {
         try
         {
             out.clear();
 
+            if (!IsValidCodePage(codepage))
+            {
+                return E_INVALIDARG;
+            }
+
             if (in.empty())
             {
                 return S_OK;
             }
 
             int lengthIn{};
+            RETURN_HR_IF(E_ABORT, !base::MakeCheckedNum(in.length()).AssignIfValid(&lengthIn));
+
             int lengthRequired{};
-            // Code Point U+0000..U+FFFF: 1 UTF-16 code unit --> 1..3 UTF-8 code units.
-            // Code Points >U+FFFF: 2 UTF-16 code units --> 4 UTF-8 code units.
-            // Thus, the worst ratio of UTF-16 code units to UTF-8 code units is 1 to 3.
-            RETURN_HR_IF(E_ABORT, !base::MakeCheckedNum(in.length()).AssignIfValid(&lengthIn) || !base::CheckMul(lengthIn, 3).AssignIfValid(&lengthRequired));
-            out.resize(gsl::narrow_cast<size_t>(lengthRequired)); // avoid to call WideCharToMultiByte twice only to get the required size
-            const int lengthOut = WideCharToMultiByte(gsl::narrow_cast<UINT>(CP_UTF8), 0ul, in.data(), lengthIn, out.data(), lengthRequired, nullptr, nullptr);
+            if (codepage == gsl::narrow_cast<unsigned int>(CP_UTF8)) // UTF-8
+            {
+                // Avoid to call WideCharToMultiByte twice only to get the required size.
+                // Code Point U+0000..U+FFFF: 1 UTF-16 code unit --> 1..3 UTF-8 code units.
+                // Code Points >U+FFFF: 2 UTF-16 code units --> 4 UTF-8 code units.
+                // Thus, the worst ratio of UTF-16 code units to UTF-8 code units is 1 to 3.
+                RETURN_HR_IF(E_ABORT, !base::CheckMul(lengthIn, 3).AssignIfValid(&lengthRequired));
+            }
+            else // ANSI codepages
+            {
+#pragma prefast(suppress \

```

Comment:
Instead of suppressing twice in quick succession.... maybe
```
#pragma warning(push)
#pragma warning(disable: __WARNING_W2A_BEST_FIT, "...")
// function body
#pragma warning(pop)
```
around the whole function?